tests/rptest/tests/cpu_profiler_admin_api_test.py

```diff
@@ -88,3 +92,64 @@ def test_get_cpu_profile_with_override_limits(self):
             pass
         else:
             assert False, "call with wait_ms < 1ms should have failed"
+
+
+class CPUProfilerStressTest(RedpandaTest):
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context,
+                         num_brokers=1,
+                         log_config=LoggingConfig(
+                             'info', logger_levels={'resources': 'trace'}),
+                         extra_rp_conf={
+                             "cpu_profiler_enabled": False,
+                             "cpu_profiler_sample_period_ms": 1,
+                         })
+
+        self.admin = Admin(self.redpanda)
+
+        self.redpanda.set_resource_settings(
+            ResourceSettings(memory_mb=4096, num_cpus=10))
+
+    @cluster(num_nodes=1)
+    def test_cpu_profile_stress(self):
+        self.redpanda.set_cluster_config({
+            "cpu_profiler_enabled": True,
+        })
+
+        node0 = self.redpanda.nodes[0]
+
+        self.admin.stress_fiber_start(num_fibers=1,
+                                      node=node0,
+                                      min_spins_per_scheduling_point=1000,
+                                      max_spins_per_scheduling_point=1000,
+                                      stack_depth=64)
+
+        # There is currently effectively a max sample count of ~1280 samples
+        # due to a limit of 10 sample buffers * 128 samples per buffer, so
+        # waiting 1,500 ms is enough for a worst-case.
+        wait_ms = 1500
+        profile = self.admin.get_cpu_profile(wait_ms=wait_ms, node=node0)
+        assert_profile_good_v2(profile)
+
+        max_samples = -1
+        total_dropped = 0
+        for shard_sample in profile["profile"]:
+            shard = shard_sample["shard_id"]
+            samples = shard_sample["samples"]
+            total_samples = 0
+            total_dropped += shard_sample["dropped_samples"]
+            for sample in samples:
+                sc = sample["user_backtrace"].count(" ") + 1
+                max_samples = max(max_samples, sc)
+                total_samples += sample["occurrences"]
+                # self.logger.warn(f"{shard} stack had {sc} samples")
+            self.logger.warn(
+                f"Shard {shard} had {len(samples)} unique samples, {total_samples} total, {shard_sample['dropped_samples']} dropped"
+            )
+
+        # CPU profiler has a max stack depth of 64 per sample, which we should
+        # hit as we set the stress fiber to use that deep of a stack (plus there
```

Comment:
Struggling to understand how that might happen as we only hit it if the seed/counter goes up all the way to 2^63 which seems unlikely? I guess there is a fair amount of stack frames already before the loop so we don't need to go as high?

Reply:
To get 64 frames in any stack you only need to make 64 recursive calls, no need for a seed at all. The thing about the seed is to ensure we have many *unique* samples, i.e., around 1200 unique samples. To do that, the seed only needs to run from 0 to 1200, as each seed gives a unique sample (as long as the number of frames is >= the number of used bits in the seed, which is always the case here).